crossbeam-skiplist/src/base.rs

```diff
@@ -2121,3 +2196,184 @@ fn below_upper_bound<T: Ord + ?Sized>(bound: &Bound<&T>, other: &T) -> bool {
         Bound::Excluded(key) => other < key,
     }
 }
+
+/// An entry where the node is ref counted in a skip list.
+///
+/// You *must* call `release` to free this type, otherwise the node will be
+/// leaked. This is because releasing the entry requires a `Guard`.
+pub struct OwnedEntry<K, V> {
+    node: *const Node<K, V>,
+    released: bool,
+}
+
+impl<K, V> OwnedEntry<K, V> {
+    /// Tries to create a new `RefCountedEntry` by incrementing the reference count of
+    /// a node.
+    fn try_acquire(node: &Node<K, V>) -> Option<OwnedEntry<K, V>> {
+        if unsafe { node.try_increment() } {
+            Some(OwnedEntry {
+                node: node as *const _,
+                released: false,
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Returns a reference to the key.
+    pub fn key(&self) -> &K {
+        unsafe { &(*self.node).key }
+    }
+
+    /// Returns a reference to the value.
+    pub fn value(&self) -> &V {
+        unsafe { &(*self.node).value }
+    }
+
+    /// Releases the reference on the entry.
+    pub fn release(mut self, guard: &Guard) {
+        self.released = true;
+        unsafe { (*self.node).decrement(guard) }
+    }
+}
+
+impl<K, V> fmt::Debug for OwnedEntry<K, V>
+where
+    K: fmt::Debug,
+    V: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("OwnedEntry")
+            .field(self.key())
+            .field(self.value())
+            .finish()
+    }
+}
+
+unsafe impl<K, V> Send for OwnedEntry<K, V> {}
+
+impl<K, V> Drop for OwnedEntry<K, V> {
+    fn drop(&mut self) {
+        assert!(self.released);
+    }
+}
+
+/// A iterator with a clone of the concurrent skip list
+pub struct OwnedIter<T, K, V>
+where
+    T: AsRef<SkipList<K, V>>,
```

Comment:
Why use a generic parameter here? There seems to be only one possible instance.

Reply:
I forgot... I think it's unnecessary now. Have removed.